security/integration/main.go

```diff
@@ -159,8 +187,86 @@ func deleteTestNamespace(clientset kubernetes.Interface) {
 	glog.Infof("Namespace %v is deleted", name)
 }
 
-func deployIstioCA(clientset kubernetes.Interface) {
-	// Create Role
+func createService(clientset kubernetes.Interface, name string, port int32,
+serviceType v1.ServiceType, pod *v1.Pod) (*v1.Service, error) {
+	uuid := string(uuid.NewUUID())
+	_, err := clientset.CoreV1().Services(opts.namespace).Create(&v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				"uuid": uuid,
+			},
+			Name: name,
+		},
+		Spec: v1.ServiceSpec{
+			Type:     serviceType,
+			Selector: pod.Labels,
+			Ports: []v1.ServicePort{
+				{
+					Port: port,
+				},
+			},
+		},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if serviceType == v1.ServiceTypeLoadBalancer {
+		err = waitForServiceExternalIPAddress(uuid, 300 * time.Second)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return clientset.CoreV1().Services(opts.namespace).Get(name, metav1.GetOptions{})
+}
+
+func createPod(clientset kubernetes.Interface, image string, name string) (*v1.Pod, error) {
+	uuid := string(uuid.NewUUID())
+
+	pod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels: map[string]string{
+				"uuid":      uuid,
+				"pod-group": fmt.Sprintf("%v-pod-group", name),
+			},
+			Name: name,
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				v1.Container{
```

Comment:
nit: could you move this container definition out from this pod definition, to avoid too many indents?

Reply:
Moved